optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -676,6 +677,9 @@ def forward(
         next_decoder_cache = () if not use_new_cache else None
 
         for layer_idx, decoder_layer in enumerate(self.layers):
+            if  torch.distributed.is_initialized() == False:
+                htcore.mark_step()
```

Comment:
@kalyanjk what's the impact for input/output not introduced oom? should we add an argument in text-generation from cmd line?

Reply:
@kalyanjk ,why only mark_step() for 1x?

Reply:
For 8x mark_step will be introduced through a collective call.

Reply:
> @kalyanjk what's the impact for input/output not introduced oom? should we add an argument in text-generation from cmd line?

The issue is not with oom. The real issue is recipe size being too large and also compilation time is too high.

Reply:
Please update as below
if lazy_mode and (torch.distributed.is_initialized() is False or torch.distributed.get_world_size() == 1):